### ISSUE TITLE
Multi login method (Standard and Social)

### ISSUE THREAD

**Issue by geoffreylzf** [django-oauth/django-oauth-toolkit#780]
Is there any standard login (by using email and password) for this library? other than social auth

**Comment by cleder** [comment]
I do not understand the question.
You log into the OAuth server provided by the DOT with username and password as usual.

**Comment by geoffreylzf** [comment]
What I mean is "by using this library, I can sign up my user account directly from my facebook or google account, and sign in it afterward using the social account",
At the same time, I also need old style way to sign up and sign in using email and password

**Comment by bastbnl** [comment]
I'm not sure if I understand your question completely, but maybe this answer helps.

Login is handled by Django and the authenticated user gets a token. It doesn't really matter if this user signed up via a local username and password or via social. 

Does this help?

**Comment by geoffreylzf** [comment]
Thank you very much, I finally grab the concept

**Comment by geoffreylzf** [comment]
Sorry to bother again, did I need to use another library(Python Social Auth) in order to authenticate with social platform like Facebook?

**Comment by bastbnl** [comment]
Yes. Or you build the Facebook integration  yourself. I used django-allauth but I've not used the Facebook provider. It's in there so you should be fine.

**Comment by matmair** [comment]
@geoffreylzf can this be closed?
